Update tests for new RDKit stereo definition

**Describe the bug**
As of the 2020.3 release, RDKit considers trivalent phosphorous to be stereogenic. This triggered one of our drugbank molecule tests (number 5329) to fail, since the molecule now is considered to have undefined stereochemistry. 

I've already updated #146 with this information. Here's the RDKit PR for reference: https://github.com/rdkit/rdkit/issues/2949

To fix this, we need to add this molecule to the `rdkit_drugbank_undefined_stereo_mols` set in `test_molecule`

**Output**
```
openforcefield/tests/test_molecule.py:283 (TestMolecule.test_to_from_smiles[molecule370-RDKitToolkitWrapper])
self = <openforcefield.tests.test_molecule.TestMolecule object at 0x12b68f890>
molecule = Molecule with name 'DrugBank_5329' and SMILES '[H]c1c(c(c(c(c1[H])[H])P2C(=C3C(=C2c4c(c(c(c(n4)[H])[H])[H])[H])C(C(C(C3([H])[H])([H])[H])([H])[H])([H])[H])c5c(c(c(c(n5)[H])[H])[H])[H])[H])[H]'
toolkit = <class 'openforcefield.utils.toolkits.RDKitToolkitWrapper'>

    @pytest.mark.parametrize('toolkit', [OpenEyeToolkitWrapper, RDKitToolkitWrapper])
    @pytest.mark.parametrize('molecule', mini_drug_bank())
    def test_to_from_smiles(self, molecule, toolkit):
        """Test round-trip creation from SMILES"""
        if not toolkit.is_available():
            pytest.skip('Required toolkit is unavailable')
    
        if toolkit == RDKitToolkitWrapper:
            # Skip the test if OpenEye assigns stereochemistry but RDKit doesn't (since then, the
            # OFF molecule will be loaded, but fail to convert in to_rdkit)
            if molecule.name in drugbank_stereogenic_in_oe_but_not_rdkit:
                pytest.skip('Molecle is stereogenic in OpenEye (which loaded this dataset), but not RDKit, so it '
                            'is impossible to make a valid RDMol in this test')
            undefined_stereo_mols = rdkit_drugbank_undefined_stereo_mols
        elif toolkit == OpenEyeToolkitWrapper:
            undefined_stereo_mols = openeye_drugbank_undefined_stereo_mols
    
        toolkit_wrapper = toolkit()
    
        undefined_stereo = molecule.name in undefined_stereo_mols
    
        smiles1 = molecule.to_smiles(toolkit_registry=toolkit_wrapper)
        if undefined_stereo:
            molecule2 = Molecule.from_smiles(smiles1,
                                             allow_undefined_stereo=True,
                                             toolkit_registry=toolkit_wrapper)
        else:
            molecule2 = Molecule.from_smiles(smiles1,
>                                            toolkit_registry=toolkit_wrapper)

test_molecule.py:312: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../topology/molecule.py:2062: in from_smiles
    allow_undefined_stereo=allow_undefined_stereo
../utils/toolkits.py:2174: in from_smiles
    allow_undefined_stereo=allow_undefined_stereo)
../utils/toolkits.py:2316: in from_rdkit
    err_msg_prefix="Unable to make OFFMol from RDMol: ")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'openforcefield.utils.toolkits.RDKitToolkitWrapper'>
rdmol = <rdkit.Chem.rdchem.Mol object at 0x12b6a20f0>
err_msg_prefix = 'Unable to make OFFMol from RDMol: ', raise_warning = False

    @classmethod
    def _detect_undefined_stereo(cls, rdmol, err_msg_prefix='', raise_warning=False):
        """Raise UndefinedStereochemistryError if the RDMol has undefined stereochemistry.
    
        Parameters
        ----------
        rdmol : rdkit.Chem.Mol
            The RDKit molecule.
        err_msg_prefix : str, optional
            A string to prepend to the error/warning message.
        raise_warning : bool, optional, default=False
            If True, a warning is issued instead of an exception.
    
        Raises
        ------
        UndefinedStereochemistryError
            If the RDMol has undefined atom or bond stereochemistry.
    
        """
        # Find undefined atom/bond stereochemistry.
        undefined_atom_indices = cls._find_undefined_stereo_atoms(rdmol)
        undefined_bond_indices = cls._find_undefined_stereo_bonds(rdmol)
    
        # Build error message.
        if len(undefined_atom_indices) == 0 and len(undefined_bond_indices) == 0:
            msg = None
        else:
            msg = err_msg_prefix + "RDMol has unspecified stereochemistry. "
            # The "_Name" property is not always assigned.
            if rdmol.HasProp("_Name"):
                msg += "RDMol name: " + rdmol.GetProp("_Name")
    
        # Details about undefined atoms.
        if len(undefined_atom_indices) > 0:
            msg += "Undefined chiral centers are:\n"
            for undefined_atom_idx in undefined_atom_indices:
                msg += ' - Atom {symbol} (index {index})\n'.format(
                    symbol=rdmol.GetAtomWithIdx(undefined_atom_idx).GetSymbol(),
                    index=undefined_atom_idx)
    
        # Details about undefined bond.
        if len(undefined_bond_indices) > 0:
            msg += "Bonds with undefined stereochemistry are:\n"
            for undefined_bond_idx in undefined_bond_indices:
                bond = rdmol.GetBondWithIdx(undefined_bond_idx)
                atom1, atom2 = bond.GetBeginAtom(), bond.GetEndAtom()
                msg += ' - Bond {bindex} (atoms {aindex1}-{aindex2} of element ({symbol1}-{symbol2})\n'.format(
                    bindex=undefined_bond_idx,
                    aindex1=atom1.GetIdx(), aindex2=atom2.GetIdx(),
                    symbol1=atom1.GetSymbol(), symbol2=atom2.GetSymbol())
    
        if msg is not None:
            if raise_warning:
                msg = 'Warning (not error because allow_undefined_stereo=True): '
                logger.warning(msg)
            else:
                msg = 'Unable to make OFFMol from RDMol: ' + msg
>               raise UndefinedStereochemistryError(msg)
E               openforcefield.utils.toolkits.UndefinedStereochemistryError: Unable to make OFFMol from RDMol: Unable to make OFFMol from RDMol: RDMol has unspecified stereochemistry. Undefined chiral centers are:
E                - Atom P (index 18)

../utils/toolkits.py:2954: UndefinedStereochemistryError
```

**Computing environment (please complete the following information):**
 - Operating system: Mac OSX
 - RDKit 2020.3

--- a/openforcefield/tests/test_molecule.py
+++ b/openforcefield/tests/test_molecule.py
@@ -208,6 +208,7 @@ rdkit_drugbank_undefined_stereo_mols = {'DrugBank_1634', 'DrugBank_1962', 'DrugB
 
 # Missing stereo in OE but not RDK:  'DrugBank_2987', 'DrugBank_3502', 'DrugBank_4161',
 # 'DrugBank_4162', 'DrugBank_6531', 'DrugBank_1700',
+drugbank_stereogenic_in_rdkit_but_not_openeye = ['DrugBank_5329']
 
 # Some molecules are _valid_ in both OETK and RDKit, but will fail if you try
 # to convert from one to the other, since OE adds stereo that RDKit doesn't
@@ -292,7 +293,7 @@ class TestMolecule:
             # Skip the test if OpenEye assigns stereochemistry but RDKit doesn't (since then, the
             # OFF molecule will be loaded, but fail to convert in to_rdkit)
             if molecule.name in drugbank_stereogenic_in_oe_but_not_rdkit:
-                pytest.skip('Molecle is stereogenic in OpenEye (which loaded this dataset), but not RDKit, so it '
+                pytest.skip('Molecule is stereogenic in OpenEye (which loaded this dataset), but not RDKit, so it '
                             'is impossible to make a valid RDMol in this test')
             undefined_stereo_mols = rdkit_drugbank_undefined_stereo_mols
         elif toolkit == OpenEyeToolkitWrapper:
@@ -301,9 +302,12 @@ class TestMolecule:
         toolkit_wrapper = toolkit()
 
         undefined_stereo = molecule.name in undefined_stereo_mols
+        # Since OpenEye did the original reading of MiniDrugBank, if OPENEYE doesn't
+        # think a feature is stereogenic, then "molecule" won't have stereochemistry defined
+        stereogenic_in_rdk_but_not_openeye = molecule.name in drugbank_stereogenic_in_rdkit_but_not_openeye
 
         smiles1 = molecule.to_smiles(toolkit_registry=toolkit_wrapper)
-        if undefined_stereo:
+        if undefined_stereo or stereogenic_in_rdk_but_not_openeye:
             molecule2 = Molecule.from_smiles(smiles1,
                                              allow_undefined_stereo=True,
                                              toolkit_registry=toolkit_wrapper)

--- a/openforcefield/tests/test_molecule.py
+++ b/openforcefield/tests/test_molecule.py
@@ -208,7 +208,7 @@ rdkit_drugbank_undefined_stereo_mols = {'DrugBank_1634', 'DrugBank_1962', 'DrugB
 
 # Missing stereo in OE but not RDK:  'DrugBank_2987', 'DrugBank_3502', 'DrugBank_4161',
 # 'DrugBank_4162', 'DrugBank_6531', 'DrugBank_1700',
-drugbank_stereogenic_in_rdkit_but_not_openeye = ['DrugBank_5329']
+drugbank_stereogenic_in_rdkit_but_not_openeye = {'DrugBank_5329'}
 
 # Some molecules are _valid_ in both OETK and RDKit, but will fail if you try
 # to convert from one to the other, since OE adds stereo that RDKit doesn't
@@ -307,7 +307,7 @@ class TestMolecule:
         stereogenic_in_rdk_but_not_openeye = molecule.name in drugbank_stereogenic_in_rdkit_but_not_openeye
 
         smiles1 = molecule.to_smiles(toolkit_registry=toolkit_wrapper)
-        if undefined_stereo or stereogenic_in_rdk_but_not_openeye:
+        if undefined_stereo or ((toolkit is RDKitToolkitWrapper) and stereogenic_in_rdk_but_not_openeye):
             molecule2 = Molecule.from_smiles(smiles1,
                                              allow_undefined_stereo=True,
                                              toolkit_registry=toolkit_wrapper)

